flash/data/data_pipeline.py

```diff
@@ -414,30 +336,39 @@ def _attach_preprocess_to_model(
                 self._model_transfer_to_device_wrapper(model.transfer_batch_to_device, device_collate_fn, model, stage)
             )
 
-    def _create_uncollate_postprocessors(self) -> _PostProcessor:
+    def _create_uncollate_postprocessors(self, stage: RunningStage) -> _PostProcessor:
         save_per_sample = None
         save_fn = None
 
+        postprocess = self._postprocess_pipeline
+
+        func_names: Dict[str, str] = {
+            k: self._resolve_function_hierarchy(k, postprocess, stage, object_type=Postprocess)
+            for k in self.POSTPROCES_FUNCS
+        }
+
         # since postprocessing is exclusive for prediction, we don't have to check the resolution hierarchy here.
-        if self._postprocess_pipeline._save_path:
-            save_per_sample = self._is_overriden('save_sample', self._postprocess_pipeline, Postprocess)
+        if postprocess._save_path:
+            save_per_sample = self._is_overriden_recursive(
```

Comment:
```suggestion
            save_per_sample: bool = self._is_overriden_recursive(
```
not sure what type is this

Reply:
Yes, bool.